pkg/controller/job/job_controller.go

```diff
@@ -705,6 +705,13 @@ func (jm *Controller) syncJob(ctx context.Context, key string) (forget bool, rEr
 	action := metrics.JobSyncActionReconciling
 
 	defer func() {
+		// Let us queue the job back accounting to the activeDeadlineSeconds. The duration after which it should be
+		// re-queued is activeDeadlineSeconds - time spent after start time.
+		if job.Spec.ActiveDeadlineSeconds != nil {
```

Comment:
This code is duplicated in line 762.

The solution is to move that code out of the if `job.Status.StartTime == nil` conditional. But it should still be conditioned on `jobSuspended(&job)`

Reply:
ack.

In the `defer` call, do you prefer if I did a `job.Status.StartTime  != nil` just to be on the safer side ?

Reply:
You don't need to do it in the defer. You can do it in line 762 after we know that `StartTime` is set.

Reply:
Cool. We are on the same page then.

Reply:
@alculquicondor I've addressed the changes suggested by you including the tests. PTAL.